plugins/inputs/socketstat/socketstat_test.go

```diff
@@ -0,0 +1,84 @@
+package socketstat
+
+import (
+	"bytes"
+	"errors"
+	"reflect"
+	"testing"
+
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSocketstat_Gather(t *testing.T) {
+	for i, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			i++
+			ss := &Socketstat{
+				SocketProto: tt.proto,
+				lister: func(cmdName string, proto string, timeout config.Duration) (*bytes.Buffer, error) {
+					return bytes.NewBufferString(tt.value), nil
+				},
+			}
+			acc := new(testutil.Accumulator)
+
+			err := ss.Init()
+			require.NoError(t, err)
+
+			err = acc.GatherError(ss.Gather)
+			if !reflect.DeepEqual(tt.err, err) {
+				t.Errorf("%d: expected error '%#v' got '%#v'", i, tt.err, err)
+			}
+			if len(tt.proto) == 0 {
+				n := acc.NFields()
+				if n != 0 {
+					t.Errorf("%d: expected 0 fields if no protocol specified got %d", i, n)
+				}
+				return
+			}
+			if len(tt.tags) == 0 {
+				n := acc.NFields()
+				if n != 0 {
+					t.Errorf("%d: expected 0 values got %d", i, n)
+				}
+				return
+			}
+			n := 0
+			for j, tags := range tt.tags {
+				for k, fields := range tt.fields[j] {
+					if len(acc.Metrics) < n+1 {
+						t.Errorf("%d: expected at least %d values got %d", i, n+1, len(acc.Metrics))
+						break
+					}
```

Comment:
```suggestion
					require.LessOrEqual(t, len(acc.Metrics), n)
```

Reply:
Pushed a fix (we actually want it to be Greater).